test/graphql/create_project_mutation_test.rb

```diff
@@ -0,0 +1,267 @@
+                                              name: 'New Project One',
+                                              description: 'New Project One Description' })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['createProject']
+
+    assert_not_empty data, 'createProject should be populated when no authorization errors'
+    assert_empty data['errors']
+    assert_not_empty data['project']
+
+    assert_equal 'New Project One', data['project']['name']
+    assert_equal 'New Project One Description', data['project']['description']
+
+    project = Project.last
+    assert_equal group.puid, project.parent.puid
+    assert_equal 'new_project_one', project.path
+    assert_equal 'group-1/new_project_one', project.full_path
+  end
+
+  test 'createProject mutation should work with valid params, no group id/puid, and api scope token' do
+    result = IridaSchema.execute(CREATE_PROJECT_MUTATION,
+                                 context: { current_user: @user, token: @api_scope_token },
+                                 variables: { name: 'New Project One',
+                                              description: 'New Project One Description' })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['createProject']
+
+    assert_not_empty data, 'createProject should be populated when no authorization errors'
+    assert_empty data['errors']
+    assert_not_empty data['project']
+
+    assert_equal 'New Project One', data['project']['name']
+    assert_equal 'New Project One Description', data['project']['description']
+
+    project = Project.last
+    assert_equal @user.namespace.puid, project.parent.puid
+    assert_equal 'new_project_one', project.path
+    assert_equal 'john.doe_at_localhost/new_project_one', project.full_path
+  end
+
+  test 'createProject mutation should work with custom path' do
+    result = IridaSchema.execute(CREATE_PROJECT_MUTATION_WITH_CUSTOM_PATH,
+                                 context: { current_user: @user, token: @api_scope_token },
+                                 variables: { name: 'New Project One',
+                                              description: 'New Project One Description',
+                                              path: 'New Custom Path' })
+
+    assert_nil result['errors'], 'should work and have no errors.'
+
+    data = result['data']['createProject']
+
+    assert_not_empty data, 'createProject should be populated when no authorization errors'
+    assert_empty data['errors']
+    assert_not_empty data['project']
+
+    assert_equal 'New Project One', data['project']['name']
+    assert_equal 'new_custom_path', data['project']['path']
+    assert_equal 'john.doe_at_localhost/new_custom_path', data['project']['fullPath']
+  end
+
+  test 'createProject mutation should not work with valid params, and api scope token with uploader access level' do # rubocop:disable Layout/LineLength
```

Comment:
rubocop warning

Reply:
Fixed in https://github.com/phac-nml/irida-next/pull/795/commits/38cb768fefd35edce0ec924190cf6e5f7b6a4a79